backend/internal/storage/postgres/assignments.go

```diff
@@ -22,15 +20,36 @@ func (db *DB) GetAllAssignments(ctx context.Context) ([]models.Assignment, error
 }
 
 func (db *DB) CreateAssignment(ctx context.Context, assignmentData models.Assignment) error {
-	_, err := db.connPool.Exec(ctx, "INSERT INTO assignments (rubric_id, semester_id, name) VALUES ($1, $2, $3)",
-		assignmentData.RubricID,
-		assignmentData.SemesterID,
-		assignmentData.Name)
-	if err != nil {
-		return err
+
+	if assignmentData.Rubric_ID == nil {
+		fmt.Println("Creating Assignment without a rubric")
+		_, err := db.connPool.Exec(ctx, "INSERT INTO assignments (inserted_date, assignment_classroom_id, semester_id, name, main_due_date) VALUES ($1, $2, $3, $4, $5)",
+			assignmentData.InsertedDate,
+			assignmentData.Assignment_Classroom_ID,
+			assignmentData.SemesterID,
+			assignmentData.Name,
+      assignmentData.MainDueDate)
+
+		if err != nil {
+			return err
+		}
+
+	} else {
+		fmt.Println("Creating Assignment with a rubric")
+		_, err := db.connPool.Exec(ctx, "INSERT INTO assignments (rubric_id, inserted_date, assignment_classroom_id, semester_id, name, main_due_date) VALUES ($1, $2, $3, $4, $5, $6)",
```

Comment:
what would happen on conflict here? If I entered the exact same information twice, would the first be overwritten, or would an error be thrown?

Reply:
discussed irl